models/index.js

```diff
@@ -1,67 +1,91 @@
+
+  return db.query(queryStr).then(({ rows }) => {
+    return rows;
+  });
+};
 
 exports.selectCommentsById = (article_id) => {
-    const regex=/\D/
-    const isNonDigit=regex.test(article_id)
-    if (article_id<=0 || isNonDigit) {
-        return Promise.reject({ status: 400, msg: "Bad request" })
-    }
-    
-    const queryStr = `SELECT * FROM comments WHERE comments.article_id = $1 ORDER BY comments.created_at DESC;`
-    
-    return db.query(queryStr,[article_id]).then(({ rows }) => {
-        return rows;
-      });
-}
+  const regex = /\D/;
+  const isNonDigit = regex.test(article_id);
+  if (article_id <= 0 || isNonDigit) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+
+  const queryStr = `SELECT * FROM comments WHERE comments.article_id = $1 ORDER BY comments.created_at DESC;`;
+
+  return db.query(queryStr, [article_id]).then(({ rows }) => {
+    return rows;
+  });
+};
 
 exports.checkArticleExists = (article_id) => {
-    const queryStr = `SELECT * FROM articles WHERE article_id = $1 ;`
-    return db.query(queryStr,[article_id]).then(({ rows }) => {
-        if(rows.length===0){
-            return Promise.reject({ status: 404, msg: "Article not found" });
-        }
-      });
-}
+  const queryStr = `SELECT * FROM articles WHERE article_id = $1 ;`;
+  return db.query(queryStr, [article_id]).then(({ rows }) => {
+    if (rows.length === 0) {
+      return Promise.reject({ status: 404, msg: "Article not found" });
+    }
+  });
+};
 
 exports.checkArticlesExists = () => {
-    const queryStr = format(`SELECT * FROM articles;`)
-    return db.query(queryStr).then(({ rows }) => {
-        if(rows.length===0){
-            return Promise.reject({ status: 404, msg: "not found" });
-        }
-      });
-    }
\ No newline at end of file
+  const queryStr = format(`SELECT * FROM articles;`);
+  return db.query(queryStr).then(({ rows }) => {
+    if (rows.length === 0) {
+      return Promise.reject({ status: 404, msg: "not found" });
+    }
+  });
+};
+
+exports.addCommentById = (newComment, article_id) => {
+  const { username, body } = newComment;
+  const regex = /\D/;
+  const isNonDigit = regex.test(article_id);
+  if (article_id <= 0 || isNonDigit) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+  if (
+    typeof username !== "string" ||
+    username.toUpperCase() === username.toLowerCase()
+  ) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+  if (typeof body !== "string" || body.toUpperCase() === body.toLowerCase()) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+  const queryStr = format(
+    `INSERT INTO comments (body, author, article_id) VALUES %L RETURNING * ;`,
+    [[body, username, article_id]]
+  );
```

Comment:
Nice use of pg-format, if you'd like to save the use of an additional dependancy (ie. not use pg-format) then you could use $ syntax here instead
If you do continue to use pg-format anywhere in your project other than the seed - make sure you move it from your devDependancies in your `package.json` to your dependancies